src/mito2/src/memtable/merge_tree/partition.rs

```diff
@@ -224,6 +226,13 @@ pub(crate) struct PartitionStats {
     pub(crate) shared_memory_size: usize,
 }
 
+#[derive(Default)]
+struct PartitionReaderMetrics {
+    prune_pk: Duration,
```

Comment:
Do we need to count number of keys before and after pruning?

Reply:
> Do we need to count number of keys before and after pruning?

fixed https://github.com/GreptimeTeam/greptimedb/pull/3397/commits/8a1736bc7931ec9ffaf44962aff46713c5f0bc91#diff-5066fcdd3b887926278b1155cfbf005cbca27e9699e36292abcc55ecf80110e2R547
not sure if these are the correct place to collect pruned keys.